lib/fluent/plugin/gcloud_pubsub/client.rb

```diff
@@ -42,22 +43,22 @@ def initialize(project, key, autocreate_topic, metric_prefix)
           Fluent::GcloudPubSub::Metrics.register_or_existing(:"#{metric_prefix}_messages_compressed_size_per_original_size_ratio") do
             ::Prometheus::Client.registry.histogram(
               :"#{metric_prefix}_messages_compressed_size_per_original_size_ratio",
-              "Compression ratio achieved on a batch of messages",
-              {},
+              docstring: "Compression ratio achieved on a batch of messages",
               # We expect compression for even a single message to be typically
               # above 2x (0.5/50%), so bias the buckets towards the higher end
               # of the range.
-              [0, 0.25, 0.5, 0.75, 0.85, 0.9, 0.95, 0.975, 1],
+              buckets: [0, 0.25, 0.5, 0.75, 0.85, 0.9, 0.95, 0.975, 1],
+              labels: [:topic, :algorithm]
             )
           end
 
         @compression_duration =
           Fluent::GcloudPubSub::Metrics.register_or_existing(:"#{metric_prefix}_messages_compression_duration_seconds") do
             ::Prometheus::Client.registry.histogram(
               :"#{metric_prefix}_messages_compression_duration_seconds",
-              "Time taken to compress a batch of messages",
-              {},
-              [0, 0.0001, 0.0005, 0.001, 0.01, 0.05, 0.1, 0.25, 0.5, 1],
+              docstring: "Time taken to compress a batch of messages",
+              buckets: [0, 0.0001, 0.0005, 0.001, 0.01, 0.05, 0.1, 0.25, 0.5, 1],
+              labels: [:topic, :algorithm]
```

Comment:
these is because of a breaking change on their side i guess?

Reply:
Yep, a former GC person made this change but we've never updated this on our side